src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -0,0 +1,97 @@
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(DoctorPatientAssignmentConstants.DOCTOR_ID_INVALID_KEY, messageSource);
+        }
+        try {
+            patientId = UUID.fromString(doctorPatientAssignmentDTO.getPatientID());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(DoctorPatientAssignmentConstants.PATIENT_ID_INVALID_KEY, messageSource);
+        }
+        if (!doctorRepository.existsById(doctorId))
+            throw new CustomException(DoctorPatientAssignmentConstants.DOCTOR_NOT_FOUND_KEY, messageSource);
+        if (!patientRepository.existsById(patientId))
+            throw new CustomException(DoctorPatientAssignmentConstants.PATIENT_NOT_FOUND_KEY, messageSource);
+    }
+
+    /**
+     * Checks if doctor is previously assigned to same patient and if not then saves the doctorPatientAssignmentDTO
+     *
+     * @param doctorPatientAssignmentDTO DTO object containing doctorId and patientId
+     * @return DoctorPatientAssignmentDTO
+     * @throws CustomException if any exception occurs
+     */
+    public DoctorPatientAssignmentDTO assignDoctorToPatient(DoctorPatientAssignmentDTO doctorPatientAssignmentDTO) throws CustomException {
+        validateDoctorPatientAssignmentDTO(doctorPatientAssignmentDTO);
+        List<DoctorPatientAssignmentModel> doctorPatientAssignmentList =
+                doctorPatientAssignmentRepository.findByDoctorIDAndPatientID
+                        (UUID.fromString(doctorPatientAssignmentDTO.getDoctorID()),
+                                UUID.fromString(doctorPatientAssignmentDTO.getPatientID()));
+        if (!doctorPatientAssignmentList.isEmpty()) {
+            for (DoctorPatientAssignmentModel doctorPatientAssignmentModel : doctorPatientAssignmentList) {
+                if (doctorPatientAssignmentModel.getDateOfUnassignment() == null) {
+                    throw new CustomException(DoctorPatientAssignmentConstants.DOCTOR_ALREADY_ASSIGNED_KEY, messageSource);
+                }
+            }
+        }
+        return mapDoctorPatientAssignmentDTOToModel(doctorPatientAssignmentDTO);
+    }
+
+    private DoctorPatientAssignmentDTO mapDoctorPatientAssignmentDTOToModel(DoctorPatientAssignmentDTO doctorPatientAssignmentDTO) {
+        DoctorPatientAssignmentDTO responseDoctorPatientAssignmentDTO;
+        DoctorPatientAssignmentModel doctorPatientAssignmentModel = DoctorPatientAssignmentModel.builder()
+                .id(UUID.randomUUID())
+                .doctorID(UUID.fromString(doctorPatientAssignmentDTO.getDoctorID()))
+                .patientID(UUID.fromString(doctorPatientAssignmentDTO.getPatientID()))
+                .dateOfAssignment(Date.from(Instant.now()))
+                .dateOfUnassignment(null)
+                .build();
+        doctorPatientAssignmentRepository.save(doctorPatientAssignmentModel);
```

Comment:
Where does assignment is happening ?
The way you are adding to the db is just an auditing operation not assignment
Also this operation should have been done with the implementation of Mongo many to many relationship between doctors and patients collection !

Reply:
We are using a separate collection for assignment so to avoid data duplicacy